### DISCUSSION TITLE
BTT Smart Sensor + E3V2 + Octopi?

### DISCUSSION THREAD

**Discussion by Stapler44** [Jyers/Marlin#1695] [Q&A]
Hello All!

I just got into 3D printing a month ago, and love it!  I have an Ender 3 V2 w/ 4.2.2 main board.  I'm running the E3V2-BLTouch-5x5-HS-v4.2.2-v2.0.1 firmware, and print using Octoprint 1.7.2 off a Pi4B.  

I've searched around for hours, but don't know what direction I should go.  I want to install the **BTT Smart Filament Sensor**, and am wondering if I should connect it to the main board or to the Pi?  If main board, will Jyers/Marlin let an Octoprint print know there's an issue, and allow me to fix it?  If I connect the BTT sensor to the Pi, will I have to use Octoprint-Smart-Filament-Sensor plugin?   What's best?  Is the configuation of the Marlin fw different depending what route I go?  I have not yet learned how to compile the firmware, but I know that'll come in handy later, so happy to dig into that too.  Thoughts?

Thank you all!

**Comment by BlueWings172** [comment]
I tried for weeks to make it work by pluging the BTT sensor it directly to the board and I auto complied firmware but it did not work. 

Someone even sent me the same firmware they use on their Ender 3 V2 with 4.2.2 board and we both got different behavior. Feel results are not consistent and my sensor is now working as a filament guide. 

Connecting it to the RPi might be a better and more stable solution but I've yet to try it.

**Reply by Stapler44** [reply]
Thank you!  Yeah, wish there was a definitive setup guide to know what the best arrangement would be.  If connected to Octoprint, then you need this plugin?  https://plugins.octoprint.org/plugins/smartfilamentsensor/

Anyone else get this to work?

Thanks!
